src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.Log.cs

```diff
@@ -0,0 +1,120 @@
+
+            private static readonly Action<ILogger, Exception> _socketClosed =
+                LoggerMessage.Define(LogLevel.Information, new EventId(2, "SocketClosed"), "Socket closed.");
+
+            private static readonly Action<ILogger, WebSocketCloseStatus?, string, Exception> _clientClosed =
+                LoggerMessage.Define<WebSocketCloseStatus?, string>(LogLevel.Debug, new EventId(3, "ClientClosed"), "Client closed connection with status code '{status}' ({description}). Signaling end-of-input to application.");
+
+            private static readonly Action<ILogger, Exception> _waitingForSend =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(4, "WaitingForSend"), "Waiting for the application to finish sending data.");
+
+            private static readonly Action<ILogger, Exception> _failedSending =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(5, "FailedSending"), "Application failed during sending. Sending InternalServerError close frame.");
+
+            private static readonly Action<ILogger, Exception> _finishedSending =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(6, "FinishedSending"), "Application finished sending. Sending close frame.");
+
+            private static readonly Action<ILogger, Exception> _waitingForClose =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(7, "WaitingForClose"), "Waiting for the client to close the socket.");
+
+            private static readonly Action<ILogger, Exception> _closeTimedOut =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(8, "CloseTimedOut"), "Timed out waiting for client to send the close frame, aborting the connection.");
+
+            private static readonly Action<ILogger, WebSocketMessageType, int, bool, Exception> _messageReceived =
+                LoggerMessage.Define<WebSocketMessageType, int, bool>(LogLevel.Debug, new EventId(9, "MessageReceived"), "Message received. Type: {messageType}, size: {size}, EndOfMessage: {endOfMessage}.");
+
+            private static readonly Action<ILogger, int, Exception> _messageToApplication =
+                LoggerMessage.Define<int>(LogLevel.Debug, new EventId(10, "MessageToApplication"), "Passing message to application. Payload size: {size}.");
+
+            private static readonly Action<ILogger, long, Exception> _sendPayload =
+                LoggerMessage.Define<long>(LogLevel.Trace, new EventId(11, "SendPayload"), "Sending payload: {size} bytes.");
+
+            private static readonly Action<ILogger, Exception> _errorWritingFrame =
+                LoggerMessage.Define(LogLevel.Error, new EventId(12, "ErrorWritingFrame"), "Error writing frame.");
+
+            private static readonly Action<ILogger, Exception> _sendFailed =
+                LoggerMessage.Define(LogLevel.Trace, new EventId(13, "SendFailed"), "Socket failed to send.");
```

Comment:
Why is this trace?

Reply:
Was trace before. Will promote to Debug.